src/coreclr/tools/aot/ILCompiler.LLVM/Compiler/LLVMCodegenCompilationBuilder.cs

```diff
@@ -70,6 +70,7 @@ public override ICompilation ToCompilation()
             {
                 case OptimizationMode.None:
                     jitFlagBuilder.Add(CorJitFlag.CORJIT_FLAG_DEBUG_CODE);
+                    jitFlagBuilder.Add(CorJitFlag.CORJIT_FLAG_DEBUG_INFO);
```

Comment:
This should be based on the command line settings and not tied to the optimization level, e.g.:
```
            // Do not bother with debug information if the debug info provider never gives anything.
            if (!(_debugInformationProvider is NullDebugInformationProvider))
                jitFlagBuilder.Add(CorJitFlag.CORJIT_FLAG_DEBUG_INFO);
```
(This can be fixed in follow-up PR.)